@vates/nbd-client/multi.mjs

```diff
@@ -0,0 +1,61 @@
+import { NBD_DEFAULT_BLOCK_SIZE } from './constants.mjs'
+import NbdClient from './index.mjs'
+
+export default class MultiNbdClient {
+  #clients = []
+  #readAhead
+
+  constructor(settings, { nbdConcurrency = 8, readAhead = 16, ...options } = {}) {
+    this.#readAhead = readAhead
+    if (!Array.isArray(settings)) {
+      settings = [settings]
+    }
+    for (let i = 0; i < nbdConcurrency; i++) {
+      this.#clients.push(
+        new NbdClient(settings[i % settings.length], { ...options, readAhead: Math.ceil(readAhead / nbdConcurrency) })
+      )
+    }
+  }
+
+  async connect() {
+    for (const client of this.#clients) {
+      await client.connect()
```

Comment:
Disconnect connected clients on error.

Reply:
I am using an alternate approach in a later commit, and continue if at least one client is successfully connected